src/js/components/InfiniteScroll/__tests__/InfiniteScroll-test.js

```diff
@@ -75,4 +76,52 @@ describe('InfiniteScroll', () => {
     );
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  test('length when step < array', () => {
+    const allItems = Array(1000)
+      .fill()
+      .map((_, i) => `item ${i + 1}`);
+    const step = 50;
+    const { container } = render(
+      <Grommet>
+        <InfiniteScroll
+          items={allItems}
+          // show={117}
+          step={step}
+        >
+          {item => <Box key={item}>{item}</Box>}
+        </InfiniteScroll>
+      </Grommet>,
+    );
+    expect(container.firstChild.children.length).toEqual(step + 1);
```

Comment:
Why `step + 1`? If I am unfamiliar with how this component is implemented, then I may not have context for why adding 1 to the step. A comment could be helpful here.

Reply:
i could be wrong, but i believe is because it'll render items 0(inclusive)-50(inclusive)? making it 51 items rendered.

Reply:
I think in this case, the 51st element is actually a 0 height ref to mark the end of the page of results.

Reply:
ah, yes. that makes more sense

Reply:
I added comments about `step + 1`. Let me know if they are too long or could be phrased better

Reply:
I think what we have here is good for now. We may revisit as we introduce additional tests and try to repurpose what is being done here.

For context, what '1' is representing here is an additional ref to serve as an anchor/trigger for additional functionality. As soon as we add a prop such as `show` there will be an additional ref marking the show index position in the middle of the resulting items and so the children in the array would then be step + 2.

@IanKBovard can you think of a way where we can test for `renderedItems.length === step`? Perhaps we do something like:
```
    const allChildren = container.firstChild.children;
    // Removing any children which are serving as refs
    const refKeys = ["first", "last", "show"];
    const pageItems = allChildren.filter( /* some sort of filter logic which returns children where there key is not in refKeys */ )
    const expectedItems = step;
    expect(pageItems.length).toEqual(expectedItems);
```

Reply:
I agree. Filter is probably the best approach.

Reply:
@jcfilben Can you take a look at the suggested approach above. If it makes sense, would you mind taking a stab at implementing?

Reply:
If possible, i think we can move all creating and filtering of children logic in a before block when we start to refactor these in future prs.

Reply:
Yup, just updated the PR using the approach outlined above